### PR TITLE
Fix messagebus subscribe on topic

### PR DESCRIPTION
**Pull request by ericclappier-eaton**

### CODE REVIEW

File: src/fty_common_messagebus_malamute.cc [hunk 1]
```diff
@@ -330,11 +330,26 @@ namespace messagebus {
...
         Message msg = _fromZmsg(message);
 
+        // FIXME: Workaround for malamute, if the author of the message is not found with the subject,
+        // then try with the address of the message which correspond to the name of the topic.
+
+        // Try first with subject
         auto iterator = m_subscriptions.find (subject);
         if (iterator != m_subscriptions.end ()) {
             //iterator->second(msg);
             std::thread (iterator->second, msg).detach();
```
Comment (perrettecl) [inline src/fty_common_messagebus_malamute.cc:340]: @boricj @MimiXav As I already said I'm not big fan of detaching the thread which execute the callback. In my opinion it's much better to give this choice to the user of the lib. 

If the user wants to detach the thread he just as to do "std::this_thread.detach() in his callback. How ever, if he want to re-synchronize to unstack the message one by one, it became more complex.

Moreover if you want to terminate the application after the last treatment, this make it difficult to as the thread will not we attach and the process can finish before, which can result to corrupted data in db for instance.

**Review by perrettecl — APPROVED** [review]

File: src/fty_common_messagebus_malamute.cc (hunk 1 above)
Comment (perrettecl) [reply src/fty_common_messagebus_malamute.cc:340]: I have a "solution" to make evrybody happy: we add a boolean "isAsynchronous" in the function to register the callback with a default value at "TRUE". This boolean decide what to do.

==>> Improvement for next time

Comment (boricj) [reply src/fty_common_messagebus_malamute.cc:340]: I would not spawn threads at all. If the client wants asynchronous processing, the client should spawn a thread or enqueue the message to a processing queue.
